homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -48,77 +27,35 @@
-    vol.Required(CONF_PASSWORD): cv.string,
-    vol.Required(CONF_USERNAME): cv.string,
-    vol.Optional(CONF_VERSION, default=DEFAULT_VERSION): cv.string,
-    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    vol.Optional(CONF_REPORT_SERVER_CODES): vol.All(cv.ensure_list),
-    vol.Optional(CONF_REPORT_SERVER_ENABLED,
-                 default=DEFAULT_REPORT_SERVER_ENABLED): cv.boolean,
-    vol.Optional(CONF_REPORT_SERVER_PORT, default=DEFAULT_REPORT_SERVER_PORT):
-        cv.port,
-})
+D_EGARDIASYS = 'egardiadevice'
+D_EGARDIADEV = 'egardia_dev'
+D_EGARDIASRV = 'egardiaserver'
+CONF_REPORT_SERVER_CODES_IGNORE = 'ignore'
+CONF_REPORT_SERVER_CODES = 'report_server_codes'
+CONF_REPORT_SERVER_ENABLED = 'report_server_enabled'
+CONF_REPORT_SERVER_PORT = 'report_server_port'
 
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Egardia platform."""
-    from pythonegardia import egardiadevice
-    from pythonegardia import egardiaserver
-
-    name = config.get(CONF_NAME)
-    username = config.get(CONF_USERNAME)
-    password = config.get(CONF_PASSWORD)
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
-    rs_enabled = config.get(CONF_REPORT_SERVER_ENABLED)
-    rs_port = config.get(CONF_REPORT_SERVER_PORT)
-    rs_codes = config.get(CONF_REPORT_SERVER_CODES)
-    version = config.get(CONF_VERSION)
-
-    try:
-        egardiasystem = egardiadevice.EgardiaDevice(
-            host, port, username, password, '', version)
-    except requests.exceptions.RequestException:
-        raise exc.PlatformNotReady()
-    except egardiadevice.UnauthorizedError:
-        _LOGGER.error("Unable to authorize. Wrong password or username")
-        return
-
-    eg_dev = EgardiaAlarm(
-        name, egardiasystem, rs_enabled, rs_codes)
-
-    if rs_enabled:
-        # Set up the egardia server
-        _LOGGER.info("Setting up EgardiaServer")
-        try:
-            if D_EGARDIASRV not in hass.data:
-                server = egardiaserver.EgardiaServer('', rs_port)
-                bound = server.bind()
-                if not bound:
-                    raise IOError(
-                        "Binding error occurred while starting EgardiaServer")
-                hass.data[D_EGARDIASRV] = server
-                server.start()
-        except IOError:
-            return
-        hass.data[D_EGARDIASRV].register_callback(eg_dev.handle_status_event)
-
-    def handle_stop_event(event):
-        """Call function for Home Assistant stop event."""
-        hass.data[D_EGARDIASRV].stop()
-
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, handle_stop_event)
-
-    add_devices([eg_dev], True)
+    device = EgardiaAlarm(
+        hass,
+        discovery_info['name'],
+        hass.data[D_EGARDIASYS],
+        discovery_info[CONF_REPORT_SERVER_ENABLED],
+        discovery_info[CONF_REPORT_SERVER_CODES] if CONF_REPORT_SERVER_CODES
```

Comment:
Use `dict.get` if you're not sure if the key exists in the dict and you're only interested in the value for the key.

Reply:
Done.